### ISSUE TITLE
[Mojave] [XQuartz] [2.8.0_rc3] Wine apps' icons and logos disappeared, connections blocked

### ISSUE THREAD

**Issue by palikao** [XQuartz/XQuartz#122]
Since the last update, 2.8.0_rc3 on macOS Mojave, Wine apps' icons and logos are no more shown inside each app, and they can't connect to internet.

**Comment by jeremyhu** [comment]
I don't see how this could be an XQuartz bug...

**Comment by palikao** [comment]
Sorry, but.. Are you crazy?! I don’t publish issues claims haphazardly..!

 

Infact, I tried reverting XQuartz to v2.7.11 and it works correctly!

 

 

 

Da: Jeremy Huddleston Sequoia ***@***.***> 
Inviato: lunedì 15 marzo 2021 16:27
A: XQuartz/XQuartz ***@***.***>
Cc: palikao ***@***.***>; Author ***@***.***>
Oggetto: Re: [XQuartz/XQuartz] [Mojave] [XQuartz] [2.8.0_rc3] Wine apps' icons and logos disappeared, connections blocked (#122)

 

I don't see how this could be an XQuartz bug...

**Comment by jeremyhu** [comment]
XQuartz has no control over the icons and logos used by wine apps.  If you think this is an XQuartz issue, you're going to need to provide more information.  Can you provide a screen capture showing what you mean?  Can you provide a means to reproduce your issue?  If so, re-open with that data, and I'll take a look.  You also might get some good help from folks on the x11-users mailing list.

**Comment by palikao** [comment]
Of course I can!!! Since I downgraded XQuartz to v2.7.11. I will provide those information in the next days, since now I need a working version for my job.

**Comment by palikao** [comment]
> Of course I can!!! Since I downgraded XQuartz to v2.7.11. I will provide those information in the next days, since now I need a working version for my job.

Here you are
![Schermata 2021-03-31 alle 16 54 54](https://user-images.githubusercontent.com/3197885/113167480-0e489000-9244-11eb-96fc-66fa085064aa.png)
![Schermata 2021-03-31 alle 16 58 24](https://user-images.githubusercontent.com/3197885/113167491-10aaea00-9244-11eb-9fce-a3de9aa15e04.png)

**Comment by jeremyhu** [comment]
Ok, can you summarize what you tried and what you saw?

What is the component that changed that introduced the problem?

Install 2.7.11, then backup XQuartz.app and /opt/X11 (ie: create a tarball of them), then update to XQuartz-2.8.0.

Extract the old XQuartz.app and run it against the new client libraries.  Does wine work?  If so, that means the issue is in the X11 server.  if not, that means it's something related to the client libraries.

What exactly?  Not sure, you'd need to do a bit more debugging to figure that out.

It's also not clear to me how Wine.app and XQuartz.app interact given that I see Wine in the menu bar rather than X11... what's that about?  How are you launching / building wine?  Have you reported this issue to the wine folks?  Can you reference that bug report here?

**Comment by palikao** [comment]
> 
> 
> Ok, can you summarize what you tried and what you saw?
In brief, I first updated (inside XQuartz or via pkg or via homebrew) and no way


> 
> What is the component that changed that introduced the problem?
I don't know, I'm here to report because I don't know.. I can only see what I screenshoted..


> 
> Install 2.7.11, then backup XQuartz.app and /opt/X11 (ie: create a tarball of them), then update to XQuartz-2.8.0.
> 
> Extract the old XQuartz.app and run it against the new client libraries. Does wine work? If so, that means the issue is in the X11 server. if not, that means it's something related to the client libraries.
Tried, no way..


> 
> What exactly? Not sure, you'd need to do a bit more debugging to figure that out.
> 
> It's also not clear to me how Wine.app and XQuartz.app interact given that I see Wine in the menu bar rather than X11... what's that about? How are you launching / building wine? Have you reported this issue to the wine folks? Can you reference that bug report here?
No, I'm not buildin, only using.. I only installed a simple Windows app (no particular need or plugin), and it works only on 2.7.11

**Comment by jeremyhu** [comment]
Ok, well please report this to the wine folks.

**Comment by Gcenx** [comment]
@palikao what version of wine where you using exactly?, I’d like to know so I can see what could be causing this as I’d managed to still support Winehq packages in Wineskin without requiring XQuartz

**Comment by Gcenx** [comment]
@palikao & @jeremyhu the above issue is related to #151 installing https://github.com/XQuartz/XQuartz/releases/tag/XQuartz-2.8.1_beta1 should get things working again.

@palikao if you had launched wine via terminal you would have seen complaints about a lack of libpng15.15.dylib, this was the cause of the issue.

The old depreciated Winehq cross-compiler was using XQuartz-2.7.7 and the MacOSX10.8.SDK.

**Comment by palikao** [comment]
> 
> 
> @palikao what version of wine where you using exactly?, I’d like to know so I can see what could be causing this as I’d managed to still support Winehq packages in Wineskin without requiring XQuartz

5.0, as WineHQ website reports it as official

**Comment by palikao** [comment]
> 
> 
> @palikao & @jeremyhu the above issue is related to #151 installing https://github.com/XQuartz/XQuartz/releases/tag/XQuartz-2.8.1_beta1 should get things working again.
> 
> @palikao if you had launched wine via terminal you would have seen complaints about a lack of libpng15.15.dylib, this was the cause of the issue.
> 
> The old depreciated Winehq cross-compiler was using XQuartz-2.7.7 and the MacOSX10.8.SDK.

Ok, in the next few days I will try it

**Comment by palikao** [comment]
Tried, and.. For now it works!
